add alternative solution (#6201)
cpinitiative/usaco-guide: solutions/silver/cses-1084.mdx

</PySection>

</LanguageSection>

# Solution 2

## Explanation

> We can use binary search to find an apartment that is in the range of

Reviewer: idk if i'd call it binary search, maybe just like using sorted set builtin methods

an applicant's desired size.

- Using lower bound, we can find the smallest apartment that has at least the
  smallest size the applicant could tolerate
- Make sure that this apartment is smaller than or equal to the
  largest size the applicant could tolerate
- Do nothing if smallest apartment with at least the smallest size is
  out of the range
- If the apartment is in the tolerable range, we remove it from the multiset
  and add one to our count. Removing it makes sure another applicant cannot take it.

This is possible with a multiset: removal is efficient and elements
may be duplicated.

Note that we have to sort the apartments so applicants could be

> matched with the apartment closed to their desired size.

Reviewer: I think it'd help to expand on this, e.g. saying smth like

> Since $k$ is fixed across all applicants, sorting the applicants by $a_i$ results in the range of acceptable apartment sizes for each applicant being sorted in increasing value. This motivates us to try pairing applicants with apartments by iterating on applicants in sorted order of $a_i$.

or something, might be dtm but rn it feels a bit arbitrary

</PySection>

</LanguageSection>

# Solution 2

## Explanation

> We can use sorted set builtin to find an apartment that is in the range of

Reviewer 2: ```suggestion
We can use the sorted set builtin to find an apartment whose size lies within an applicant's acceptable range.```

an applicant's desired size.

> - Using lower bound, we can find the smallest apartment that has at least the

Reviewer 2: ```suggestion
- Using lower bound, we find the smallest available apartment whose size is at least the minimum size the applicant can tolerate
```

  smallest size the applicant could tolerate
- Make sure that this apartment is smaller than or equal to the
  largest size the applicant could tolerate
- Do nothing if smallest apartment with at least the smallest size is
  out of the range
- If the apartment is in the tolerable range, we remove it from the multiset

>   and add one to our count. Removing it makes sure another applicant cannot take it.

Reviewer 2: ```suggestion
  and add one to our count. Removing it ensures another applicant cannot take it.
```
